backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/computing/purchasecontract/storagecost/StorageCostComputingMethod.java

```diff
@@ -106,18 +105,31 @@ public boolean applies(final @NonNull TableRecordReference recordRef, @NonNull f
 		}
 		Check.assumeEquals(request.getCurrencyId(), pricePerUnitPerDay.getCurrencyId(), "Log and Invoice Currency should be the same");
 
-		final Money storageCosts = computeStorageCosts(logs, pricePerUnitPerDay).orElseGet(() -> Money.zero(request.getCurrencyId()));
+		final Money storageCosts = computeStorageCosts(logs, pricePerUnitPerDay);
 
 		final UomId stockUOMId = productBL.getStockUOMId(request.getProductId());
-		return ComputingResponse.builder()
-				.ids(logs.getIds())
-				.price(ProductPrice.builder()
-						.productId(request.getProductId())
-						.money(storageCosts.negateIf(request.isCostInvoicingGroup()))
-						.uomId(stockUOMId)
-						.build())
-				.qty(Quantitys.one(stockUOMId))
+		final ProductPrice priceWithPriceUOM = ProductPrice.builder()
+				.productId(request.getProductId())
+				.money(storageCosts.negateIf(request.isCostInvoicingGroup()))
+				.uomId(pricePerUnitPerDay.getUomId())
 				.build();
+
+		if(UomId.equals(stockUOMId, pricePerUnitPerDay.getUomId()))
```

Comment:
is this really needed?
(i think some of the conversation methods are already checking)

Reply:
It's a different product uom (settingline one)

Reply:
nvm got it changed